OpenRA.Game/Manifest.cs

```diff
@@ -174,5 +179,29 @@ static string[] YamlList(Dictionary<string, MiniYaml> yaml, string key, bool par
 
 			return module;
 		}
+
+		static Dictionary<string, Manifest> LoadMods()
+		{
+			var basePath = Platform.ResolvePath(".", "mods");
+			var mods = Directory.GetDirectories(basePath)
+				.Select(x => x.Substring(basePath.Length + 1));
+
+			var ret = new Dictionary<string, Manifest>();
+			foreach (var mod in mods)
+			{
+				try
+				{
+					var manifest = new Manifest(mod);
+					ret.Add(mod, manifest);
+				}
+				catch (Exception ex)
+				{
+					Log.Write("debug", "An exception occured while trying to load mod {0}:", mod);
+					Log.Write("debug", ex.ToString());
```

Comment:
We could throw an exception with more relevant information.
This is fine for now, but I'd like to revisit this in the future.
Possibly with a new exception just for this.